packages/application-extension/src/index.tsx

```diff
@@ -243,10 +236,11 @@ const mainCommands: JupyterFrontEndPlugin<void> = {
           return;
         }
         const { id } = widget;
-        const otherWidgets = toArray(shell.widgets('main')).filter(
-          widget => widget.id !== id
-        );
-        closeWidgets(otherWidgets);
+        for (const widget of shell.widgets('main')) {
```

Comment:
We can mark an untouched line. But if you look up at line 231, the condition is redundant :wink: 

And it seems that some of the remaining error are due to call to `next()` not followed by a test on `done`.

Reply:
The condition wasn't redundant, but it *was* wrong for not checking `done`. It specifically wants to make sure there are at least **two** widgets:
```ts
// Ensure there are at least two widgets.
const iterator = shell.widgets('main');
return !iterator.next().done && !iterator.next().done;
```

Reply:
I went with this so it won't look like a mistake:
```ts
// Ensure there are at least two widgets.
return some(shell.widgets('main'), (_, i) => i === 1);
```